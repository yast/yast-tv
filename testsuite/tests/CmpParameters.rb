# encoding: utf-8

module Yast
  class CmpParametersClient < Client
    def main
      Yast.import "UI"
      # testedfiles: Tv.ycp tv/misc.ycp

      Yast.include self, "testsuite.rb"

      @READ = { "target" => { "size" => -1, "tmpdir" => "/tmp" } }

      TESTSUITE_INIT([@READ], nil)

      Yast.import "Tv" # moves "Read    .target.size "/etc/install.inf" -1" line up
      Yast.include self, "tv/misc.rb"


      TEST(lambda { CmpParameters({}, {}) }, [], nil)

      TEST(lambda { CmpParameters({ "card" => "0" }, { "card" => "0" }) }, [], nil)

      TEST(lambda do
        CmpParameters(
          { "card" => "0", "tuner" => "0" },
          { "card" => "0", "tuner" => "1" }
        )
      end, [], nil)

      TEST(lambda do
        CmpParameters({ "card" => "0", "tuner" => nil }, { "card" => "0" })
      end, [], nil)

      nil
    end
  end
end

Yast::CmpParametersClient.new.main
