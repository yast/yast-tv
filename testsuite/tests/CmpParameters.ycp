{
    // testedfiles: Tv.ycp tv/misc.ycp

    include "testsuite.ycp";

    map READ = $[
	"target" : $[ "size": -1, "tmpdir" : "/tmp", ]
    ];

    TESTSUITE_INIT([READ], nil);

    import "Tv"; // moves "Read    .target.size "/etc/install.inf" -1" line up
    include "tv/misc.ycp";


	TEST(``(CmpParameters($[],$[])), [], nil);

	TEST(``(CmpParameters($[ "card":"0" ],$[ "card":"0" ])), [], nil);

	TEST(``(CmpParameters($[ "card":"0", "tuner":"0" ],$[ "card":"0", "tuner": "1" ])), [], nil);

	TEST(``(CmpParameters($[ "card":"0", "tuner":nil ],$[ "card":"0" ])), [], nil);

}
