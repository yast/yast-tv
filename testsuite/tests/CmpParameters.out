Read	.target.size "/etc/install.inf" -1
Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp nil
Read	.probe.system nil
Return	true
Return	true
Return	false
Return	true
